sda/cmd/ingest/ingest_test.go

```diff
@@ -102,9 +130,14 @@ func (suite *TestSuite) TestTryDecrypt() {
 	buf, err := io.ReadAll(file)
 	assert.NoError(suite.T(), err)
 
-	key, err := config.GetC4GHKey()
-	assert.NoError(suite.T(), err)
-	header, err := tryDecrypt(key, buf)
+	keyList, err := config.GetC4GHprivateKeys()
 	assert.NoError(suite.T(), err)
-	assert.NotNil(suite.T(), header)
+	for _, key := range keyList {
+		header, err := tryDecrypt(key, buf)
+		if header != nil && err == nil {
+			break
+		}
+		assert.NoError(suite.T(), err)
+		assert.NotNil(suite.T(), header)
+	}
```

Comment:
```suggestion
	for i, key := range keyList {
		header, err := tryDecrypt(key, buf)
		switch {
		case i == 1:
			assert.NoError(suite.T(), err)
			assert.NotNil(suite.T(), header)
		default:
			assert.EqualError(suite.T(), "expected error string",err)
			assert.Nil(suite.T(), header)
		}
	}
```

If you generate a set of different c4gh keys and want to test that only one can decrypt the file you need to write the test something like this.

Reply:
Fixed in b81ce66c